File: telegram-ui/miniapps.ts
/**
 * Telegram Mini App (WebApp) button URLs — placeholders from env/config.
 */

export interface MiniAppUrls {
  dashboard: string;
  terminal: string;
  portfolio: string;
  research: string;
  agents: string;
  settings: string;
}

export const MINIAPP_ENV_KEYS = {
  dashboard: "PMX_TELEGRAM_MINIAPP_DASHBOARD",
  terminal: "PMX_TELEGRAM_MINIAPP_TERMINAL",
  portfolio: "PMX_TELEGRAM_MINIAPP_PORTFOLIO",
  research: "PMX_TELEGRAM_MINIAPP_RESEARCH",
  agents: "PMX_TELEGRAM_MINIAPP_AGENTS",
  settings: "PMX_TELEGRAM_MINIAPP_SETTINGS",
} as const;

export function loadMiniAppUrls(env: Record<string, string | undefined>): MiniAppUrls {
  return {
    dashboard: env.PMX_TELEGRAM_MINIAPP_DASHBOARD?.trim() || "https://example.com/pmx/dashboard",
    terminal: env.PMX_TELEGRAM_MINIAPP_TERMINAL?.trim() || "https://example.com/pmx/terminal",
    portfolio: env.PMX_TELEGRAM_MINIAPP_PORTFOLIO?.trim() || "https://example.com/pmx/portfolio",
    research: env.PMX_TELEGRAM_MINIAPP_RESEARCH?.trim() || "https://example.com/pmx/research",
    agents: env.PMX_TELEGRAM_MINIAPP_AGENTS?.trim() || "https://example.com/pmx/agents",
    settings: env.PMX_TELEGRAM_MINIAPP_SETTINGS?.trim() || "https://example.com/pmx/settings",
  };
}

export const MINIAPP_LABELS: Record<keyof MiniAppUrls, string> = {
  dashboard: "Open Dashboard",
  terminal: "Trading Terminal",
  portfolio: "Portfolio",
  research: "Research Board",
  agents: "Agent Control Center",
  settings: "Settings",
};

File: apps/telegram/ui/miniapps.py
"""Mini App URLs from env — mirrors telegram/miniapps.ts."""

from __future__ import annotations

import os
from dataclasses import dataclass

from apps.telegram.ui.loader import load_ui_spec


@dataclass(frozen=True)
class MiniAppUrls:
    dashboard: str
    terminal: str
    portfolio: str
    research: str
    agents: str
    settings: str


def load_mini_app_urls() -> MiniAppUrls:
    spec = load_ui_spec()["miniApps"]
    env_keys: dict[str, str] = spec["envKeys"]
    defaults: dict[str, str] = spec["defaults"]
    values = {
        key: os.environ.get(env_keys[key], "").strip() or defaults[key]
        for key in env_keys
    }
    return MiniAppUrls(**values)


def mini_app_labels() -> dict[str, str]:
    return dict(load_ui_spec()["miniApps"]["labels"])
